Tests/Microsoft.AppCenter.Crashes.Test.Windows/HandledErrorTest.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AppCenter.Channel;
+using Microsoft.AppCenter.Crashes.Ingestion.Models;
+using Microsoft.AppCenter.Ingestion.Http;
+using Microsoft.AppCenter.Ingestion.Models;
+using Microsoft.AppCenter.Ingestion.Models.Serialization;
+using Microsoft.AppCenter.Storage;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.AppCenter.Crashes.Test.Windows
+{
+    [TestClass]
+    public class HandledErrorTest
+    {
+        private static IHttpNetworkAdapter StartCrashes()
+        {
+            var mockHttpNetworkAdapter = Mock.Of<IHttpNetworkAdapter>();
+            var storage = new Storage.Storage(new StorageAdapter("test.db"));
+            var ingestion = new IngestionHttp(mockHttpNetworkAdapter);
+            var channelGroup = new ChannelGroup(ingestion, storage, "app secret");
+            Crashes.Instance = new Crashes();
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(channelGroup, "app secret");
+            return mockHttpNetworkAdapter;
+        }
+
+        [TestMethod]
+        public void TrackErrorWithValidParameters()
+        {
+            var mockHttpNetworkAdapter = StartCrashes();
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            Mock.Get(mockHttpNetworkAdapter).Setup(adapter => adapter.SendAsync(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<IDictionary<string, string>>(),
+                It.IsAny<string>(),
+                It.IsAny<CancellationToken>()))
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    actualLog = JsonConvert.DeserializeObject<LogContainer>(content, LogSerializer.SerializationSettings).Logs.Single() as HandledErrorLog;
+                    semaphore.Release();
+                });
+            var exception = new System.Exception("Something went wrong.");
+            var properties = new Dictionary<string, string> { { "k1", "v1" }, { "p2", "v2" } };
+            Crashes.TrackError(exception, properties);
+
+            // Wait until the http layer sends the log.
+            semaphore.Wait(1000);
```

Comment:
Can we refactor an internal method that returns a Task (or assigns a Task to a property)? This is going to be flaky at some points. I also thought about this when I implemented `ProcessPendingErrorsAsync`. `ProcessPendingErrorsTask` property is the one that I added for testing purpose.
Or, is this something that we can resolve in functional test? Then I'm fine.

Reply:
I'm not sure the solution will work here. `Crashes` can get a `Task` for the channel's `EnqueueAsync` and expose it to the test, but that task returns once the log is stored to disk. It doesn't wait until the crash was sent to ingestion. Thus, we would need to significantly refactor the `Channel`. So there is no really good way to avoid the timeout.

Reply:
Then is 1 sec wait enough? Channel is configured to send logs every 1 sec if there are logs. It should be waiting more than 1 sec but I don't think it is worth to address it here or discuss more about this one since we have a plan to change it already. 😄